acceptancetests/jujupy/client.py

```diff
@@ -2122,6 +2133,80 @@ def switch(self, model=None, controller=None):
         self.juju('switch', (':'.join(args),), include_e=False)
 
 
+# caas `add-k8s` did not implement parsing kube config path via flag,
+# so parse it via env var ->  https://github.com/kubernetes/client-go/blob/master/tools/clientcmd/loader.go#L44
+KUBE_CONFIG_PATH_ENV_VAR = 'KUBECONFIG'
+
+
+class CaasClient(ModelClient):
+
+    cloud_name = 'k8cloud'
+    kubectl_path = None
+    kube_home = None
+    kube_config_path = None
+
+    def __prepare(self):
+        # to pick up the new tmp juju_home, have to do this!
+        self.kubectl_path = os.path.join(self.env.juju_home, 'kubectl')
+        self.kube_home = os.path.join(self.env.juju_home, '.kube')
+        self.kube_config_path = os.path.join(self.kube_home, 'config')
+
+    def cluster_up(self, bundle_path):
+        self.__prepare()
+
+        self.deploy_bundle(bundle_path, static_bundle=True)
+        # Wait for the deployment to finish.
+        self.wait_for_started()
+
+        # ensure kube home
+        ensure_dir(self.kube_home)
+        # ensure kube credentials
+        self.juju('scp', ('kubernetes-master/0:config', self.kube_config_path))
+
+        # ensure kubectl by scp from master
+        self.juju('scp', ('kubernetes-master/0:/snap/bin/kubectl', self.kubectl_path))
+
+        # ensure kube config env var
+        os.environ[KUBE_CONFIG_PATH_ENV_VAR] = self.kube_config_path
+        # ensure k8s cloud
+        self.add_caas_cloud()
+        log.debug('added caas cloud, now all clouds are -> \n%s', self.list_clouds())
+
+    def add_caas_cloud(self):
+        self.juju('add-k8s', self.cloud_name)
+
+    def add_caas_model(self, caas_model_name):
+        return self.add_model(env=self.env.clone(caas_model_name), cloud_region=self.cloud_name)
+
+    @property
+    def is_cluster_healthy(self):
+        try:
+            cluster_info = self._kubectl_cmd('cluster-info')
+            log.debug('cluster_info -> \n%s', cluster_info)
+            nodes_info = self._kubectl_cmd('get nodes')
+            log.debug('nodes_info -> \n%s', nodes_info)
+            return True
+        except RuntimeError:
+            return False
+
+    def _kubectl_cmd(self, args):
+        args = args if isinstance(args, list) else [args]
+        args = [sub_flag for flag in args for sub_flag in flag.split(' ') if sub_flag]
+        args = [self.kubectl_path, '--kubeconfig={}'.format(self.kube_config_path)] + args
+        return self._sh(args)
+
+    def _sh(self, cmd):
+        cmd = cmd if isinstance(cmd, list) else [cmd]
+        cmd_str = ' '.join(cmd)
+        log.debug('_sh: cmd -> `%s`, \n\tcmd_splitted -> %s', cmd, cmd_str)
+        exitcode, data = subprocess.getstatusoutput(cmd_str)
```

Comment:
sorry, @Veebers  `getstatusoutput` will only work for `py3`, 
Are we going to run this job in py3 or py2?